torch/export/dynamic_shapes.py

```diff
@@ -1046,8 +1069,12 @@ def _get_dim_name_mapping(
         dynamic_shapes,
         is_leaf=lambda x: isinstance(x, _Dim),
     )[0]:
-        if isinstance(dim, (int, _DimHint)) or dim is None:
+        if dim is None:
```

Comment:
do we not warn on tensor-level Nones?

Reply:
So the invariant I'm shooting for here is that things corresponding to input tree leaves can be `None` (both tensors and things like constants), but tensor dimensions cannot (because they can be STATIC).